feat: add `/releases` page

amrbashir opened the pull request:
<!-- Thank you for opening a PR! We really appreciate you taking the time to help out 🙌 -->

#### What kind of changes does this PR include?
<!-- Delete any that don’t apply -->
<!-- Translators, try to follow this pattern when naming your PRs:
"i18(language code): (short description)" -->

- New or updated content
- Changes to the docs site code

#### Description
<!-- Delete any that don’t apply -->
- What does this PR change? Give us a brief description. <!-- If it's an update try adding the commits as reference -->
Add a release notes page.

<!--
Here’s what will happen next:

1. Our GitHub bots will run to check your changes.
   If they spot any broken links you will see some error messages on this PR.
   Don’t hesitate to ask any questions if you’re not sure what these mean!

2. In a few minutes, you’ll be able to see a preview of your changes on Netlify 🥳

3. One or more of our maintainers will take a look and may ask you to make changes.
   We try to be responsive, but don’t worry if this takes a day or two.

4. Reach out to us on Discord with any questions along the way: 
   https://discord.com/invite/tauri
-->

amrbashir committed:
feat: add `/releases` page

lorenzolewis commented:
I'm really curious if we need this when the same information is on GitHub. What's the objective you're wanting from this? Just a more focused and filtered list of releases to highlight the more important/popular packages?

amrbashir commented:
That and Github releases page UX is awful and doesn't support quick navigation to a certain release (unless you know exactly what release you want)

amrbashir committed:
Merge branch 'next' into feat/releases

lorenzolewis commented:
Hmm, that's a good point. I also wonder if change logs for the plugins would be nice to have? That could be included on the respective plugin pages.  At first I was a bit torn at having the change details inline and running into formatting issues, but it's all markdown so that shouldn't be an issue. 

On the implementation, I'd prefer polling the GitHub API during build and statically have the changelog on pages instead of dynamically on each page load. That'd reduce overall page load time for the viewer and make it feel snappier (plus makes it so we have less client-side JS to ship).

I'd like to play around with the UI a little bit, but what are your thoughts on the above before that?

amrbashir commented:
> Hmm, that's a good point. I also wonder if change logs for the plugins would be nice to have? That could be included on the respective plugin pages. At first I was a bit torn at having the change details inline and running into formatting issues, but it's all markdown so that shouldn't be an issue.

Nice one, but inlining is probably not the way to go, just to not pollute documentation, I think all releases just be under `/releases`, because ideally we want to support `tauri.app/releases/tauri/1.0.2` (which I just thought of now, so I will convert this PR to a draft until I get some time to get back to ti)

> On the implementation, I'd prefer polling the GitHub API during build and statically have the changelog on pages instead of dynamically on each page load. That'd reduce overall page load time for the viewer and make it feel snappier (plus makes it so we have less client-side JS to ship)

That particular page doesn't need to have the best performance but yeah that would probably be better as it will make implementing `tauri.app/releases/tauri/1.0.2` easier.
 
> I'd like to play around with the UI a little bit, but what are your thoughts on the above before that?

Definitely feel free to do so, I don't very much like the styling I did

lorenzolewis commented:
Cool. If we do that then I can also just update some of the plugin meta links on the pages to point to a changelog page. Give me a shout if you want me to take a look at anything in the meantime.

amrbashir committed:
Merge branch 'next' into feat/releases

amrbashir committed:
remove solid

amrbashir committed:
generated on build instead

amrbashir committed:
update lockfile

amrbashir commented:
Shifted this PR to generate the releases pages on build, should be ready for review.

amrbashir commented:
I don't know why build fails with links validation tbh, @lorenzolewis do you have any idea why this happens?

lorenzolewis commented:
I believe because the links are formatted like `/releases/@tauri-apps/cli/v2_0_0-alpha_18` but astro with change those to `/releases/tauri-apps/cli/v2_0_0-alpha_18` to get rid of any special characters (note the `@` is removed). They use https://www.npmjs.com/package/github-slugger under the hood (or a custom impl that gets the same output) so I'd use that to generate any links (and possibly any filenames as well).

lorenzolewis commented on src/content/docs/releases/index.mdx:
```diff
@@ -0,0 +1,12 @@
+template: splash
+---
+
+- [`tauri`](/releases/tauri)
+- [`@tauri-apps/api`](/releases/tauri)
+- [`tauri-cli`](/releases/tauri)
+- [`@tauri-apps/cli`](/releases/tauri)
+- [`tauri-bundler`](/releases/tauri)
+- [`wry`](/releases/tauri)
+- [`tao`](/releases/tauri)
```
I think these links need to be updated, right?

amrbashir replied:
nice catch

lorenzolewis committed:
Merge remote-tracking branch 'origin/next' into feat/releases

Signed-off-by: Lorenzo Lewis <lorenzo_lewis@icloud.com>

amrbashir committed:
update links

amrbashir committed:
Update index.mdx

amrbashir commented:
> I believe because the links are formatted like `/releases/@tauri-apps/cli/v2_0_0-alpha_18` but astro with change those to `/releases/tauri-apps/cli/v2_0_0-alpha_18` to get rid of any special characters (note the `@` is removed). They use [npmjs.com/package/github-slugger](https://www.npmjs.com/package/github-slugger) under the hood (or a custom impl that gets the same output) so I'd use that to generate any links (and possibly any filenames as well).

Can we override this behavior somehow? ideally the link should contain the `@` just fine, similar to https://paka.dev/npm/@tauri-apps/api

lorenzolewis commented:
Hmm, tbh I don't see anything in Starlight that would affect this, and generating a normal Astro page seems to retain special characters. We're using a slightly old version due to a few dependency incompatibilities so let me check in the morning if I can figure out whats happening.

lorenzolewis commented:
Actually, it may have to do with Content Collections because they do a tad bit of magic sauce on top of the default route handling. We can try this and see if it works (away from computer right now): https://docs.astro.build/en/guides/content-collections/#defining-custom-slugs

amrbashir committed:
Update build.ts

amrbashir committed:
Merge branch 'next' into feat/releases

amrbashir committed:
fix slug creations

amrbashir committed:
Add nav links back

amrbashir commented:
> Actually, it may have to do with Content Collections because they do a tad bit of magic sauce on top of the default route handling. We can try this and see if it works (away from computer right now): [docs.astro.build/en/guides/content-collections/#defining-custom-slugs](https://docs.astro.build/en/guides/content-collections/#defining-custom-slugs)

It didn't work unfortunately, the error is coming from this plugin though https://github.com/HiDeoo/starlight-links-validator

lorenzolewis commented:
I'm thinking this is caused by this regex upstream: https://github.com/HiDeoo/starlight-links-validator/blob/e4d378d7484995b50a9a6f5f22a098bdd597cf70/packages/starlight-links-validator/libs/remark.ts#L146

We're not on the latest version due to version incompatibilities with the blog plugin. There are a few changes in this area of the link validator in later versions. Let me message HiDeoo and see what he recommends since he maintains both plugins. I'm assuming the blog plugin isn't going to be updated immediately so I may just quickly throw something together to fix that and clear that blocker. 

Will reply back here with the feedback.

lorenzolewis commented:
Most likely will be after the holiday season as the author is indeed wanting to push through https://github.com/withastro/starlight/pull/1175 in order to unlock the other issues (and thus letting us upgrade all of our deps). I'm thinking to work through a few other things on my list and then just rolling a custom blog impl since Astro's Content Collections make it fairly trivial tbh (and can let us add RSS as well).

vasfvitor committed:
Merge branch 'next' into feat/releases

vasfvitor commented:
seeing if/how we can unblock this from upstream

vasfvitor committed:
update links valitor to 0.6.0

amrbashir commented:
thanks @vasfvitor, glad to see you have resolved the issues, I hadn't have enough time work on this lately but let me know if I can be of any help.

vasfvitor commented:
@amrbashir I was just writing this up😄. What do you need to be done to finish this one? I can get it going, just point me generally

![image](https://github.com/tauri-apps/tauri-docs/assets/61759797/69453c38-eeff-4785-aeba-99a004a68900)

amrbashir commented:
Haha, I guess "great minds think alike"

I don't think I have more to add to this PR atm so feel free to merge or add anything if you feel there is something missing.

vasfvitor committed:
Merge branch 'next' into feat/releases

vasfvitor committed:
use link card

vasfvitor committed:
wip - generated frontmatter

vasfvitor committed:
wip

vasfvitor committed:
fix labels name

vasfvitor committed:
improve logic

vasfvitor committed:
Merge branch 'next' into feat/releases

vasfvitor commented:
Added navigation: next-prev buttons, and "return to: index"

<img src="https://github.com/tauri-apps/tauri-docs/assets/61759797/00046f15-8214-4e9c-88b6-977fd5eda782" width="60%"/>


"All releases"
<img src="https://github.com/tauri-apps/tauri-docs/assets/61759797/d6a8caf1-5e55-41bc-a76c-4dd512b181b5" width="60%"/>


Card links @simonhyll idea
<img src="https://github.com/tauri-apps/tauri-docs/assets/61759797/b2e7949b-fe71-4dba-bf1e-da740bcd6082" width="60%"/>

still some stuff todo

vasfvitor committed:
WIP refactor

show link to GitHub
wip version side navigation
improve wording and navigation
move releases to references

vasfvitor committed:
add entitify

vasfvitor committed:
agora vai

vasfvitor committed:
Update build.ts

vasfvitor committed:
fix: dir name

vasfvitor committed:
add (Rust)

vasfvitor committed:
fix dirName build

vasfvitor committed:
change if order

vasfvitor committed:
remove dirName

vasfvitor committed:
redo sidenav

vasfvitor committed:
place /releases back on header

vasfvitor committed:
Merge branch 'next' into feat/releases

vasfvitor committed:
wip mobile

vasfvitor committed:
improve mobile nav

vasfvitor committed:
Update ReleaseSidebar.astro

vasfvitor committed:
Merge branch 'feat/releases' of https://github.com/tauri-apps/tauri-docs into feat/releases

simonhyll commented:
If we can get the sidebar to show on all pages I'd say we can merge this and think of improvements later.

Shows here: https://deploy-preview-1688--tauri-docs-starlight.netlify.app/releases/
Doesn't show here: https://deploy-preview-1688--tauri-docs-starlight.netlify.app/releases/tauri/v1.2.2/

vasfvitor committed:
refactor: sidebar nav + dropdown only

vasfvitor committed:
fix width and redo button width

vasfvitor commented:
@simonhyll to account for this then I changed the design so the list of releases is enclosed in a dropdown like menu, as if it's mobile. Not quite good looking imo, but gtg I'd say

simonhyll reviewed (approved):
While I would want the right sidebar to show up in a manner more similar to the existing Starlight right sidebar, I think what we have currently is good enough to merge. We can toy around with the exact appearance down the line.

vasfvitor committed:
revert to show sidebar list + hide toc

lucasfernog committed:
fix labels

lucasfernog committed:
fix sort

lucasfernog reviewed (approved):
I agree with Simon but it's tricky to generate a custom table of contents view, so let's go with this for now.

lucasfernog reviewed (approved):
I agree with Simon but it's tricky to generate a custom table of contents view, so let's go with this for now.

lucasfernog committed:
fix split

lucasfernog committed:
fix split
